components/game/Game.tsx

```diff
@@ -93,6 +153,18 @@ function Game() {
       }
     }
 
+    function touchRandomTreasures(children) {
+      children.disableBody(true, true);
+    }
+
+    function touchNecklace(necklace) {
+      necklace.disableBody(true, true);
+    }
+
+    function touchChalice(chalice) {
+      chalice.disableBody(true, true);
+    }
```

Comment:
Right now, these functions are very similar. I expect that you add differences later (like different points depending on treasure)?
If not, I would reduce it to a single function.